samples/snippets/noxfile.py

```diff
@@ -0,0 +1,310 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import glob
+import os
+from pathlib import Path
+import sys
+from typing import Callable, Dict, List, Optional
+
+import nox
+
+# WARNING - WARNING - WARNING - WARNING - WARNING
+# WARNING - WARNING - WARNING - WARNING - WARNING
+#           DO NOT EDIT THIS FILE EVER!
+# WARNING - WARNING - WARNING - WARNING - WARNING
+# WARNING - WARNING - WARNING - WARNING - WARNING
+
+BLACK_VERSION = "black==22.3.0"
+ISORT_VERSION = "isort==5.10.1"
+
+# Copy `noxfile_config.py` to your directory and modify it instead.
```

Comment:
Does this repo contain a noxfile_config.py file? I couldn't find one.

Reply:
Nope, this repo doesn't yet contain a noxfile. Hence, I imported it from other python repos.

Reply:
Can you check with yoshi-python team, to try and discover why this repo doesn't have the config file? IIRC that is part of their template, and it would be better to have the template controlled by their templating engine for ease of future maintenance.

Reply:
Also, to be clear: This file is named `noxfile.py`, but the comment refers to `noxfile_config.py`.

Reply:
This file is in fact templated. If you modify it here, and then run the owlbot post processor, the changes will be reverted by owlbot. I believe this specific templated file is only added in folders where there is a `requirements.txt`. See the source code for synthtool here: See https://github.com/googleapis/synthtool/blob/master/synthtool/languages/python.py#L141. This function is called from `owlbot.py` file here: https://github.com/googleapis/python-iam/blob/main/owlbot.py#L83

Reply:
If you'd like to propose changes to this file, please open a pull request on the synthtool templates where the templated files reside: https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/python_samples/noxfile.py.j2

Reply:
I'm not sure what the right change is, so I filed https://github.com/googleapis/synthtool/issues/1489.